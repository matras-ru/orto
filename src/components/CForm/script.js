import { selfInstall } from '@/utils';

import ThemeClass from '@/themes/default/CForm';

const { baseClass } = ThemeClass;

export default {
    name: 'Form',

    install(Vue, theme) {
        selfInstall(Vue, theme, this);
    },

    props: {
        action: {
            type: String,
            default: null
        },
        method: {
            type: String,
            default: null
        }
    },

    methods: {
        onSubmit() {
            this.$emit('submit');
        }
    },

    render(h) {
        const attrs = {
            action: this.action,
            method: this.method
        };

        const on = {
            submit: this.onSubmit
        };

        const componentData = {
            class: baseClass,
            attrs: attrs,
            on
        };

        return h('form', componentData, this.$slots.default);
    }
};
