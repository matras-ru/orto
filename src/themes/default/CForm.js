export const base = 'mb-1-4';

export default {
    base
};
