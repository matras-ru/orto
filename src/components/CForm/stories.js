import { storiesOf } from '@storybook/vue';

import CForm from './';

const stories = storiesOf('CForm', module);

stories.add('Form', () => ({
    components: { CForm },
    template: '<CForm action="/" method="POST">Form here</CForm>'
}));
