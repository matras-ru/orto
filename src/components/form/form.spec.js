/* eslint-env jest */
import { mount } from '@vue/test-utils';
import CForm from './form';
const baseClass = 'mb-1-4';

describe('Form', () => {
    it('is rendered', async () => {
        const wrapper = mount(CForm);

        expect(wrapper.element.tagName).toEqual('FORM');
        expect(wrapper.classes().sort()).toEqual(`${baseClass}`.split(' ').sort());

        expect(wrapper.html()).toMatchSnapshot();

        wrapper.destroy();
    });

    it('default slot', async () => {
        const wrapper = mount(CForm, {
            slots: {
                default: 'foobar'
            }
        });

        expect(wrapper.text()).toEqual('foobar');

        wrapper.destroy();
    });
});
