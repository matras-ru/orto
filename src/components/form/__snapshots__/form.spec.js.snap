// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form is rendered 1`] = `<form class="mb-1-4"></form>`;
